src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -155,12 +155,31 @@
                  (str/join "', '" (sort valid-methods)) "'"))))
   (when (= "regex" (-> rule :match-request :type name str/lower-case))
     (try
-      (re-pattern (:path (:match-request rule)))
+      (let [pattern (re-pattern (:path (:match-request rule)))
+            group-count (-> pattern (.matcher "") .groupCount)
+            get-br-list (fn [match-str]
+                          (map #(Integer/parseInt (nth % 1))
+                               (re-seq #"\$(\d)" match-str)))
```

Comment:
I think this is missing a `+` - `#"\$(\d+)"`  This would help catch the error for backreferences larger than 9.